### PR TITLE
replication controllers and services

### PR DESCRIPTION
**Pull request by vellichor**

### CODE REVIEW

File: deployments/production/spark-master-controller.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
+{
+    "apiVersion": "extensions/v1beta1",
+    "kind": "Deployment",
+    "metadata": {
+        "generation": 3,
```
Comment (vellichor) [inline deployments/production/spark-master-controller.json]: Is this (or anything else) cluster-side metadata that we should remove?

Comment (gabe-ochoa) [reply deployments/production/spark-master-controller.json]: @vellichor let's remove it. You're absolutely correct; it's cluster-side metadata.

**Review by gabe-ochoa — CHANGES_REQUESTED** [review]
These are different from the running configs. Let's pull the configs off the cluster and commit those.

**Review by gabe-ochoa — APPROVED** [review]
approved, but let's remove the generation metadata

File: deployments/production/spark-master-service.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        "name": "spark-cluster-master",
+        "namespace": "spark-cluster-production",
+        "resourceVersion": "3578"
+    },
+    "spec": {
+        "clusterIP": "None",
+        "ports": [
+            {
+                "name": "worker-port",
+                "nodePort": 30031,
```
Comment (gabe-ochoa) [inline deployments/production/spark-master-service.json]: Let's remove the nodeport so the cluster can auto assign.

File: deployments/production/spark-master-service.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {
+        "name": "spark-cluster-master",
+        "namespace": "spark-cluster-production",
+        "resourceVersion": "3578"
```
Comment (gabe-ochoa) [inline deployments/production/spark-master-service.json]: resourceVersion is cluster metadata

File: deployments/production/spark-master-service.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            {
+                "name": "worker-port",
+                "nodePort": 30031,
+                "port": 7077,
+                "protocol": "TCP",
+                "targetPort": 7077
+            },
+            {
+                "name": "webui",
+                "nodePort": 31193,
```
Comment (gabe-ochoa) [inline deployments/production/spark-master-service.json]: same here

File: deployments/production/spark-webui-service.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {
+        "name": "spark-webui",
+        "namespace": "spark-cluster-production",
+        "resourceVersion": "3624"
```
Comment (gabe-ochoa) [inline deployments/production/spark-webui-service.json]: - resourceVersion

File: deployments/production/spark-webui-service.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    "spec": {
+        "clusterIP": "10.3.0.97",
```
Comment (gabe-ochoa) [inline deployments/production/spark-webui-service.json]: - clusteIP

File: deployments/production/spark-webui-service.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        "ports": [
+            {
+                "nodePort": 32441,
```
Comment (gabe-ochoa) [inline deployments/production/spark-webui-service.json]: - nodeport

File: deployments/production/spark-webui-service.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+                "port": 8080,
+                "protocol": "TCP",
+                "targetPort": 8080
+            }
+        ],
+        "selector": {
+            "component": "spark-cluster-master"
+        },
+        "sessionAffinity": "None",
```
Comment (gabe-ochoa) [inline deployments/production/spark-webui-service.json]: - sessionAffinity

File: deployments/staging/spark-master-service.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
+{
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {
+        "name": "spark-cluster-master",
+        "namespace": "spark-cluster",
+        "resourceVersion": "14375877"
```
Comment (gabe-ochoa) [inline deployments/staging/spark-master-service.json]: - resourceVersion

File: deployments/staging/spark-master-service.json [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    "spec": {
+        "clusterIP": "None",
+        "ports": [
+            {
+                "name": "worker-port",
+                "nodePort": 31565,
```
Comment (gabe-ochoa) [inline deployments/staging/spark-master-service.json]: - nodePort

File: deployments/staging/spark-master-service.json [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+                "port": 7077,
+                "protocol": "TCP",
+                "targetPort": 7077
+            },
+            {
+                "name": "webui",
+                "nodePort": 31093,
```
Comment (gabe-ochoa) [inline deployments/staging/spark-master-service.json]: - nodePort

File: deployments/staging/spark-master-service.json [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+                "port": 8080,
+                "protocol": "TCP",
+                "targetPort": 8080
+            }
+        ],
+        "selector": {
+            "component": "spark-cluster-master"
+        },
+        "sessionAffinity": "None",
```
Comment (gabe-ochoa) [inline deployments/staging/spark-master-service.json]: - sessionAffinity

File: deployments/staging/spark-webui-service.json [hunk 13]
```diff
@@ -0,0 +1,25 @@
+{
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {
+        "name": "spark-webui",
+        "namespace": "spark-cluster",
+        "resourceVersion": "48587"
```
Comment (gabe-ochoa) [inline deployments/staging/spark-webui-service.json]: - resourceVersion

File: deployments/staging/spark-webui-service.json [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    "spec": {
+        "clusterIP": "10.3.0.67",
```
Comment (gabe-ochoa) [inline deployments/staging/spark-webui-service.json]: - clusterIP

File: deployments/staging/spark-webui-service.json [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+        "ports": [
+            {
+                "nodePort": 32148,
```
Comment (gabe-ochoa) [inline deployments/staging/spark-webui-service.json]: - nodePort

File: deployments/staging/spark-webui-service.json [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+                "port": 8080,
+                "protocol": "TCP",
+                "targetPort": 8080
+            }
+        ],
+        "selector": {
+            "component": "spark-cluster-master"
+        },
+        "sessionAffinity": "None",
```
Comment (gabe-ochoa) [inline deployments/staging/spark-webui-service.json]: -sessionAffinity

File: deployments/staging/spark-worker-service.json [hunk 17]
```diff
@@ -0,0 +1,25 @@
+{
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {
+        "name": "worker-service",
+        "namespace": "spark-cluster",
+        "resourceVersion": "3986596"
```
Comment (gabe-ochoa) [inline deployments/staging/spark-worker-service.json]: - resourceVersion

File: deployments/staging/spark-worker-service.json [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+                "name": "http",
+                "port": 8081,
+                "protocol": "TCP",
+                "targetPort": 8081
+            }
+        ],
+        "selector": {
+            "component": "spark-worker"
+        },
+        "sessionAffinity": "None",
```
Comment (gabe-ochoa) [inline deployments/staging/spark-worker-service.json]: - sessionAffinity

**Review by gabe-ochoa — CHANGES_REQUESTED** [review]

File: deployments/production/spark-master-controller.json [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+                            {
+                                "name": "SPARK_MASTER_PORT",
+                                "value": "7077"
+                            },
+                            {
+                                "name": "SPARK_MASTER_MEM",
+                                "value": "4G"
+                            }
+                        ],
+                        "image": "quay.io/winkapp/spark-standalone:1.6.0",
```
Comment (vellichor) [inline deployments/production/spark-master-controller.json]: @gabe-ochoa looks like the cluster is still running this branch image. Do you remember why? Should we move this to master?

Comment (gabe-ochoa) [reply deployments/production/spark-master-controller.json]: @vellichor I don't but yeah let's move it to `master` now that the work is merged.

**Review by gabe-ochoa — APPROVED** [review]

**Pull request merged**